test/functional/sendheaders.py

```diff
@@ -132,97 +161,79 @@ def on_headers(self, conn, message):
             message.headers[-1].calc_sha256()
             self.last_blockhash_announced = message.headers[-1].sha256
 
-    # Test whether the last announcement we received had the
-    # right header or the right inv
-    # inv and headers should be lists of block hashes
+    def clear_last_announcement(self):
+        with mininode_lock:
+            self.block_announced = False
+            self.last_message.pop("inv", None)
+            self.last_message.pop("headers", None)
+
     def check_last_announcement(self, headers=None, inv=None):
```

Comment:
nit: Are the default values required for `headers` and `inv`? For the case of `inv_node`, we must only ever send `inv`s and `headers==[]`. For the case of `test_node`, we should send `headers` and `inv==[]`s or `inv` and `headers==[]`. Though, it is not required for the protocol, so feel free to leave as is.

Reply:
Right, having the default arguments like this allows looser or stricter testing. If the function is called without both arguments, then the testing is looser. We can make it tighter by explicitly calling with one of the arguments set to an empty list.

I think it's ok as is, but I'm also happy to change if you have a better suggestion.